integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/template/endpointTemplate/EndpointTemplateSuspensionTest.java

```diff
@@ -0,0 +1,125 @@
+                new DataHandler(new URL("file:" + File.separator + File.separator + getESBResourceLocation() +
+                        File.separator + "car" + File.separator + "templateEndpointInRegistryTestCapp_1.0.0.car")));
+
+        boolean cAppDeployed = Utils.isCarFileDeployed(cAppName, applicationAdminClient, 120000);
+        Assert.assertTrue(cAppDeployed, "CApp templateEndpointInRegistryTestCapp_1.0.0.car deployment failed");
+    }
+
+
+    @Test(groups = "wso2.esb", description = "Test invoking EP which cause timeout and verify whether Endpoint "
+            + "template has been marked for suspension")
+    public void testTemplateEndpointSuspension () throws IOException, LogViewerLogViewerException,
+            InterruptedException {
+
+        String contentType = "text/xml";//Content-Type
+        Map<String, String> headers = new HashMap<String, String>();//For HTTP Headers
+        headers.put("Content-Type", contentType);
+        headers.put("SOAPAction", "urn:mediate");
+        String payload =    "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\">\n" +
+                            "   <soapenv:Header/>\n" +
+                            "   <soapenv:Body>\n" +
+                            "       <test>test payload</test>\n" +
+                            "   </soapenv:Body>\n" +
+                            "</soapenv:Envelope>";
+
+        LogViewerClient logViewerClient = new LogViewerClient(contextUrls.getBackEndUrl(),getSessionCookie());
+        logViewerClient.clearLogs();
+
+        log.info("Invoke testEndpointTemplateProxy attempt 1");
+        SimpleHttpClient httpClient1 = new SimpleHttpClient();
+        HttpResponse response1 = httpClient1.doPost(getProxyServiceURLHttp("testEndpointTemplateProxy"), headers, payload, contentType);
+
+        //Wait for timeout markForSuspension retry delay
+        try {
+            Thread.sleep(12000);
+        } catch (InterruptedException e) {
+            //ignore
+        }
+
+        log.info("Invoke testEndpointTemplateProxy attempt 2");
+        SimpleHttpClient httpClient2 = new SimpleHttpClient();
+        HttpResponse response2 = httpClient2.doPost(getProxyServiceURLHttp("testEndpointTemplateProxy"), headers, payload, contentType);
+
+        //Wait for timeout markForSuspension retry delay
+        try {
+            Thread.sleep(12000);
+        } catch (InterruptedException e) {
+            //ignore
+        }
+
+        log.info("Invoke testEndpointTemplateProxy attempt 3");
+        SimpleHttpClient httpClient3 = new SimpleHttpClient();
+        HttpResponse response3 = httpClient3.doPost(getProxyServiceURLHttp("testEndpointTemplateProxy"), headers, payload, contentType);
+
+        //Check logs to verify endpoint suspension
+        boolean epSuspended = Utils.checkForLog(logViewerClient,
+                "testEP1 with address http://localhost:8480/testbe has been marked for SUSPENSION", 10);
```

Comment:
Can we introduce a variable without hard coding the port here?

Reply:
We have already hardcoded above URL in test artifacts (in car file). Avoiding that hardcoding is inevitable. That URL is logged in the EI console log, hence using constant or parameterizing here does not give any advantage.